packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -108,47 +108,39 @@ export class PluginDeployerImpl implements PluginDeployer {
         await this.deployMultipleEntries([pluginEntry]);
     }
 
-    protected async deployMultipleEntries(pluginEntries: string[]): Promise<void> {
-
-        const deployedPlugins = new Set<string>();
-
-        /**
-         * Iterate over all the plugins, resolving them one at a time and adding
-         * in extension dependencies
-         */
-        const futurePlugins = [];
-        while (pluginEntries.length !== 0) {
-            const currentPlugin = pluginEntries.pop() as string;
-            if (deployedPlugins.has(currentPlugin)) {
+    protected async deployMultipleEntries(pluginEntries: ReadonlyArray<string>): Promise<void> {
+        const visited = new Set<string>();
+        const pluginsToDeploy = new Map<string, PluginDeployerEntry>();
+
+        const queue = [...pluginEntries];
+        while (queue.length) {
+            const current = queue.shift()!;
+            if (visited.has(current)) {
                 continue;
             }
+            visited.add(current);
 
             // resolve plugins
-            const pluginDeployerEntries = await this.resolvePlugin(currentPlugin);
+            const pluginDeployerEntries = await this.resolvePlugin(current);
 
             // now that we have plugins check if we have File Handler for them
             await this.applyFileHandlers(pluginDeployerEntries);
 
             // ok now ask for directory handlers
             await this.applyDirectoryFileHandlers(pluginDeployerEntries);
 
-            // add current plugin deployer entries first because dependencies to be installed first
```

Comment:
@JPinkney It does not matter in which order plugins are deployed. They are only loaded when a frontend is connected by then all plugins should be already on the disk.